controllers/numaresourcesoperator_controller.go

```diff
@@ -53,14 +53,15 @@ const (
 // NUMAResourcesOperatorReconciler reconciles a NUMAResourcesOperator object
 type NUMAResourcesOperatorReconciler struct {
 	client.Client
-	Log          logr.Logger
-	Scheme       *runtime.Scheme
-	Platform     platform.Platform
-	APIManifests apimanifests.Manifests
-	RTEManifests rtemanifests.Manifests
-	Helper       *deployer.Helper
-	Namespace    string
-	ImageSpec    string
+	Log                 logr.Logger
+	Scheme              *runtime.Scheme
+	Platform            platform.Platform
+	APIManifests        apimanifests.Manifests
+	RTEManifests        rtemanifests.Manifests
+	InitialRTEManifests rtemanifests.Manifests
```

Comment:
we will probably need manifests for each host group we define in the API (cc @cynepco3hahue )

Reply:
yep, I am currently thinking about how to do it.
Like in general under expected state we should have a number of resources as only a single one should exist under the cluster and some can have multiple instances. Something like
### Single
- service account
- role
- role binding
- scc

### Multiple
- daemon set
- machine config
- config map

Reply:
the `deployer`'s `manifest` package should offer enough flexibility for us to compose the functionality the way we need, should not that be the case we can totally send PRs to `deployer` to enable this degree of flexibility.